### ISSUE TITLE
Keymap.json doesn't get loaded(Terminal application)

### ISSUE THREAD

**Issue by Brn9hrd7** [laurent22/joplin#2360]
I've tried changing the key bindings by downloading a keymap.json file from a dotfile repo I found, but it doesn't seem to get loaded at all. 
This is the content of ~/.config/joplin/keymap.json
```
[
	{ "keys": [":"], "type": "function", "command": "enter_command_line_mode" },
	{ "keys": ["l", "RIGHT", "p"], "type": "function", "command": "focus_next" },
	{ "keys": ["h", "LEFT"], "type": "function", "command": "focus_previous" },
	{ "keys": ["k", "UP"], "type": "function", "command": "move_up" },
	{ "keys": ["j", "DOWN"], "type": "function", "command": "move_down" },
	{ "keys": ["PAGE_UP"], "type": "function", "command": "page_up" },
	{ "keys": ["PAGE_DOWN"], "type": "function", "command": "page_down" },
	{ "keys": ["ENTER", "i"], "type": "function", "command": "activate" },
	{ "keys": ["DELETE", "x", "d"], "type": "function", "command": "delete" },
    	{ "keys": ["q"], "type": "exec", "command": "exit"},
	{ "keys": [" "], "command": "todo toggle $n" },
	{ "keys": ["t"], "command": "todo clear $n" },
	{ "keys": ["c"], "type": "function", "command": "toggle_console" },
	{ "keys": ["md"], "type": "function", "command": "toggle_metadata" },
	{ "keys": ["/"], "type": "prompt", "command": "search \"\"", "cursorPosition": -2 },
	{ "keys": ["mn"], "type": "prompt", "command": "mknote \"\"", "cursorPosition": -2 },
	{ "keys": ["mt"], "type": "prompt", "command": "mktodo \"\"", "cursorPosition": -2 },
	{ "keys": ["mb"], "type": "prompt", "command": "mkbook \"\"", "cursorPosition": -2 },
	{ "keys": ["y"], "type": "prompt", "command": "cp $n \"\"", "cursorPosition": -2 },
	{ "keys": ["mv"], "type": "prompt", "command": "mv $n \"\"", "cursorPosition": -2 },
	{ "keys": ["rn"], "type": "prompt", "command": "ren $c \"\"", "cursorPosition": -2 }
]
```
## Environment

Joplin version: 1.0.178-1
Platform: Linux, terminal
OS specifcs: Arch Linux, I've installed Joplin through the AUR

## Steps To Reproduce

1. Create a keymap.json file
2. Launch joplin

I've also tried modifying the default keymap.json file, but that didn't change anything either.

Describe what you expected to happen:
I expected different key bindings after adding the keymap.json file.

## Logfile

```
2019-12-09 23:23:09: "Profile directory: /home/bernhard/.config/joplin"
2019-12-09 23:23:10: "First start: detected locale as de_DE"
2019-12-09 23:23:10: "Trying to load 0 master keys..."
2019-12-09 23:23:10: "Loaded master keys: 0"
2019-12-09 23:23:10: "ResourceFetcher: Auto-add resources: Mode: always"
2019-12-09 23:23:10: "ResourceFetcher: Auto-added resources: 0"
2019-12-09 23:23:10: "Running migration: 20"
2019-12-09 23:23:10: "Trying to load 0 master keys..."
2019-12-09 23:23:10: "Loaded master keys: 0"
2019-12-09 23:23:10: "Scheduling sync operation..."
2019-12-09 23:23:10: "Trying to load 0 master keys..."
2019-12-09 23:23:10: "Loaded master keys: 0"
2019-12-09 23:23:10: "Scheduling sync operation..."
2019-12-09 23:23:10: "Trying to load 0 master keys..."
2019-12-09 23:23:10: "Loaded master keys: 0"
2019-12-09 23:23:10: "Scheduling sync operation..."
2019-12-09 23:23:10: "RevisionService::runInBackground: Starting background service with revision collection interval 600000"
2019-12-09 23:23:11: "DecryptionWorker: cannot start because no master key is currently loaded."
2019-12-09 23:23:14: "RevisionService::maintenance: Starting..."
2019-12-09 23:23:14: "RevisionService::maintenance: Service is enabled"
2019-12-09 23:23:14: "RevisionService::collectRevisions: Created revisions for 0 notes"
2019-12-09 23:23:14: "RevisionService::maintenance: Done in 102ms"
2019-12-09 23:23:20: "SearchEngine: Updating FTS table..."
2019-12-09 23:23:20: "Preparing scheduled sync"
2019-12-09 23:23:20: "Saving updated Dropbox auth."
2019-12-09 23:23:20: "Synchroniser is missing credentials - manual sync required to authenticate."
2019-12-09 23:23:51: "Profile directory: /home/bernhard/.config/joplin"
2019-12-09 23:23:51: "Trying to load 0 master keys..."
2019-12-09 23:23:51: "Loaded master keys: 0"
2019-12-09 23:23:51: "ResourceFetcher: Auto-add resources: Mode: always"
2019-12-09 23:23:51: "ResourceFetcher: Auto-added resources: 0"
2019-12-09 23:23:52: "Trying to load 0 master keys..."
2019-12-09 23:23:52: "Loaded master keys: 0"
2019-12-09 23:23:52: "Scheduling sync operation..."
2019-12-09 23:23:52: "Trying to load 0 master keys..."
2019-12-09 23:23:52: "Loaded master keys: 0"
2019-12-09 23:23:52: "Scheduling sync operation..."
2019-12-09 23:23:52: "Trying to load 0 master keys..."
2019-12-09 23:23:52: "Loaded master keys: 0"
2019-12-09 23:23:52: "Scheduling sync operation..."
2019-12-09 23:23:52: "RevisionService::runInBackground: Starting background service with revision collection interval 600000"
2019-12-09 23:23:53: "DecryptionWorker: cannot start because no master key is currently loaded."
2019-12-09 23:23:56: "RevisionService::maintenance: Starting..."
2019-12-09 23:23:56: "RevisionService::maintenance: Service is enabled"
2019-12-09 23:23:56: "RevisionService::collectRevisions: Created revisions for 0 notes"
2019-12-09 23:23:56: "RevisionService::maintenance: Done in 102ms"
2019-12-09 23:24:01: "SearchEngine: Updating FTS table..."
2019-12-09 23:24:01: "SearchEngine: Updated FTS table in 1ms. Inserted: 0. Deleted: 0"
2019-12-09 23:24:02: "Preparing scheduled sync"
2019-12-09 23:24:02: "Saving updated Dropbox auth."
2019-12-09 23:24:02: "Synchroniser is missing credentials - manual sync required to authenticate."
2019-12-09 23:24:22: "ResourceService::indexNoteResources: Start"
2019-12-09 23:24:22: "ResourceService::indexNoteResources: Completed"
2019-12-09 23:24:22: "ResourceService::deleteOrphanResources:", "[]"
2019-12-10 02:59:34: "Profile directory: /home/bernhard/.config/joplin"
2019-12-10 02:59:34: "Trying to load 0 master keys..."
2019-12-10 02:59:34: "Loaded master keys: 0"
2019-12-10 02:59:34: "ResourceFetcher: Auto-add resources: Mode: always"
2019-12-10 02:59:34: "ResourceFetcher: Auto-added resources: 0"
2019-12-10 03:02:34: "Profile directory: /home/bernhard/.config/joplin"
2019-12-10 03:02:34: "Trying to load 0 master keys..."
2019-12-10 03:02:34: "Loaded master keys: 0"
2019-12-10 03:02:34: "ResourceFetcher: Auto-add resources: Mode: always"
2019-12-10 03:02:34: "ResourceFetcher: Auto-added resources: 0"
2019-12-10 03:02:34: "Trying to load 0 master keys..."
2019-12-10 03:02:34: "Loaded master keys: 0"
2019-12-10 03:02:34: "Scheduling sync operation..."
2019-12-10 03:02:34: "Trying to load 0 master keys..."
2019-12-10 03:02:34: "Loaded master keys: 0"
2019-12-10 03:02:34: "Scheduling sync operation..."
2019-12-10 03:02:34: "Trying to load 0 master keys..."
2019-12-10 03:02:34: "Loaded master keys: 0"
2019-12-10 03:02:34: "Scheduling sync operation..."
2019-12-10 03:02:34: "RevisionService::runInBackground: Starting background service with revision collection interval 600000"
2019-12-10 03:02:35: "DecryptionWorker: cannot start because no master key is currently loaded."
2019-12-10 03:02:38: "RevisionService::maintenance: Starting..."
2019-12-10 03:02:38: "RevisionService::maintenance: Service is enabled"
2019-12-10 03:02:38: "RevisionService::collectRevisions: Created revisions for 0 notes"
2019-12-10 03:02:38: "RevisionService::maintenance: Done in 101ms"
```

**Comment by bedwardly-down** [comment]
I would first definitely try to uninstall the AUR version, backup your Joplin config folders (.joplin and .joplin-desktop), and then download and run the latest AppImage from the [releases page](https://github.com/laurent22/joplin/releases) or, if you want the CLI version, build it from source. 

If you go the AppImage route, make sure you have `fuse2` installed or the app won't start. If you build the CliClient from source, you'll need to make sure you have nodejs and npm installed with Pacman. Then do the following:

1. [Build The Tools](https://github.com/laurent22/joplin/blob/master/BUILD.md#building-the-tools)
2. `cd CliClient`
3. `sh build.sh`
4. And if no errors occur, `sh run.sh`

That should get you started. 

The reason I'm not recommending using AUR for installing Joplin is oftentimes AUR packages mess with configs or do some weird stuff that break things. That, and I want to rule out that it's a Joplin issue and not just AUR bugging.

**Comment by Brn9hrd7** [comment]
When running `build.sh` I didn't get any output at all(not sure if this is expected).
When trying to run run.sh afterwards I got this error:
```
internal/modules/cjs/loader.js:983
  throw err;
  ^

Error: Cannot find module 'lib/JoplinServerApi.js'
Require stack:
- /home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/lib/SyncTargetNextcloud.js
- /home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/lib/BaseApplication.js
- /home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/app.js
- /home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/main.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Module.require (internal/modules/cjs/loader.js:1040:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/lib/SyncTargetNextcloud.js:9:25)
    at Module._compile (internal/modules/cjs/loader.js:1151:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1171:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Module.require (internal/modules/cjs/loader.js:1040:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/lib/SyncTargetNextcloud.js',
    '/home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/lib/BaseApplication.js',
    '/home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/app.js',
    '/home/bernhard/Build/joplin-cli-v1.0.153/CliClient/build/main.js'
  ]
}

```

**Comment by tessus** [comment]
@Brn9hrd7 Did you run `npm install` in the `CliClient` directory?

Also, `run.sh` invokes `build.sh`, so running `build.sh` by itself is not necessary.

**Comment by Brn9hrd7** [comment]
@tessus 
Yes I did, I just followed the instructions [here](https://github.com/laurent22/joplin/blob/master/BUILD.md#building-the-tools) (only the Building the tools & Building the terminal application part)

**Comment by tessus** [comment]
Just ran through it and I didn't get the error. Maybe @laurent22 has an idea.

Although build issues should be discussed on the forum. Now the build issue distracts from the original issue you've posted.

**Comment by bedwardly-down** [comment]
Sorry for asking him about that. AUR (it's like Ubuntus PPA) on Arch has a tendency to break things so i wanted to make sure it wasn't that.

**Comment by bedwardly-down** [comment]
@Brn9hrd7 i just had a thought. What window manager / dedktop environment are you currently using? Could it be possible that the keymapping settings there could be overriding your Keymap.json file's mappings? Ironically, i just ran into that issue on my iPhone with how my text replacement settings were being overridden by another one of my apps

**Comment by Brn9hrd7** [comment]
I'm using bspwm(with sxhkd), but I just tried different WM's and DE's as well(Openbox, Xfce, Qtile), and I'm experiencing this issue on all of them.

**Comment by Brn9hrd7** [comment]
Figured it out, and it was entirely my fault :D Sry for wasting your time.
A few weeks ago I decided to try out the CLI version(was using the Electron app), but then I saw that all my notes are apparently stored in ~/.config/joplin-desktop while the CLI only loads ~/.config/joplin
So I've put this line in my zshrc
`
alias joplin="joplin --profile ~/.config/joplin-desktop"
`
so all my notes get loaded. But then I continued to use the Electron App because I didn't like the default key bindings of the terminal app, and completely forgot that I've created that alias. That's kinda embarrassing, how could I forget about that... Sry again @bedwardly-down @tessus
